src/vmm/src/device_manager/persist.rs

```diff
@@ -128,6 +157,9 @@ pub struct DeviceStates {
     /// Balloon device state.
     #[version(start = 2, ser_fn = "balloon_serialize")]
     pub balloon_device: Option<ConnectedBalloonState>,
+    /// Mmds version.
+    #[version(start = 3, ser_fn = "mmds_version_serialize")]
+    pub mmds_version: Option<MmdsVersionState>,
```

Comment:
Why can't we use `MmdsVersion` here?

Reply:
We could, but we've used this pattern in the past so that a potential change in the `MmdsVersion` struct does not necessarily mean an update in the snapshot format and a version bump. For example, this is what we do for `CacheTypeState`, `FileEngineTypeState`, `BalloonStatsState` and many more structs